buildenv/jenkins/common/build.groovy

```diff
@@ -720,4 +734,49 @@ def build_all() {
     }
 }
 
+def match_fail_pattern(outputLines) {
+    if (!FAIL_PATTERN) {
+        return false
+    }
+
+    println("Build failure, searching fail pattern in the last ${outputLines.size()} output lines")
+    Pattern pattern = Pattern.compile(FAIL_PATTERN)
+    for (line in outputLines) {
+        Matcher matcher = pattern.matcher(line)
+        if (matcher.find()) {
+            return true
+        }
+    }
+
+    println("Fail pattern not found!")
+    return false
+}
+
+def recompile(startTime) {
```

Comment:
Did we want to do something with startTime, such as print it out?

Reply:
removed